compute_snapshot datasource added (#6527)
GoogleCloudPlatform/magic-modules: mmv1/third_party/terraform/website/docs/d/compute_snapshot.html.markdown

---
subcategory: "Compute Engine"
page_title: "Google: google_compute_snapshot"
description: |-
  Get information about a Google Compute Snapshot.
---

# google\_compute\_snapshot

> Get information about a Google Compute Persistent disks.

Reviewer: This seems like a copy-paste error, should it refer to snapshots?

Author: Fixed.

> [the official documentation](https://cloud.google.com/compute/docs/disks/create-snapshots) and its [API](https://cloud.google.com/compute/docs/reference/rest/v1/snapshots).

Author: fixed.

## Example Usage

```hcl
#by name 
data "google_compute_snapshot" "snapshot" {
  name    = "generic-tpl-20200107"
}

# using a filter
data "google_compute_snapshot" "latest-snapshot" {
  filter      = "name != generic-tpl-20200107"
  most_recent = true
}
```

## Argument Reference

The following arguments are supported:

* `name` - (Optional) The name of the instance template. One of `name` or `filter` must be provided.

> - `filter` - (Optional) A filter to retrieve the instance templates.

Author: Fixed.

    See [gcloud topic filters](https://cloud.google.com/sdk/gcloud/reference/topic/filters) for reference.
    If multiple instance templates match, either adjust the filter or specify `most_recent`. One of `name` or `filter` must be provided.

- `most_recent` - (Optional) If `filter` is provided, ensures the most recent template is returned when multiple instance templates match. One of `name` or `filter` must be provided.

- - -

* `project` - (Optional) The ID of the project in which the resource belongs.
    If it is not provided, the provider project is used.

> ## Attributes Reference

Author: fixed.

---
subcategory: "Compute Engine"
page_title: "Google: google_compute_snapshot"
description: |-
  Get information about a Google Compute Snapshot.
---

# google\_compute\_snapshot

To get more information about Snapshot, see:

* [API documentation](https://cloud.google.com/compute/docs/reference/rest/v1/snapshots)
* How-to Guides
    * [Official Documentation](https://cloud.google.com/compute/docs/disks/create-snapshots)

## Example Usage

```hcl
#by name 
data "google_compute_snapshot" "snapshot" {
  name    = "my-snapshot"
}

# using a filter
data "google_compute_snapshot" "latest-snapshot" {
  filter      = "name != my-snapshot"
  most_recent = true
}
```

## Argument Reference

The following arguments are supported:

* `name` - (Optional) The name of the compute snapshot. One of `name` or `filter` must be provided.

* `filter` - (Optional) A filter to retrieve the compute snapshot.
    See [gcloud topic filters](https://cloud.google.com/sdk/gcloud/reference/topic/filters) for reference.
    If multiple compute snapshot match, either adjust the filter or specify `most_recent`. One of `name` or `filter` must be provided.

> * `most_recent` - (Optional) If `filter` is provided, ensures the most recent snapshot is returned when multiple compute snapshot match. One of `name` or `filter` must be provided.

Reviewer: ```suggestion
* `most_recent` - (Optional) If `filter` is provided, ensures the most recent snapshot is returned when multiple compute snapshot match. Otherwise, this value is ignored.
```
